### DISCUSSION TITLE
Running G6 and G7 at the same time breaks Companion Bluetooth feature

### DISCUSSION THREAD

**Discussion by blaqone** [NightscoutFoundation/xDrip#2383] [General]
For testing purposes im running the G6 and G7 app at the same time. Two seperate xdrip version, one reads data from G6 (BYODA) and the other try to read the G7 via Companion Bluetooth feature.
Enabling this feature does not read the data from the 7, but instead from the 6. After disabling Companion Bluetooth it reads it from 7 again.

**Comment by Navid200** [comment]
Under normal circumstances, there is only one sensor.
If the user cannot use xDrip as a collector, they can use the Companion App feature.  In that case, there will be no issue because there will be only one collector app on the phone.

If you believe there is merit in having two sensors at once, meaning if you could see yourself telling your child to insert two sensors at once, please explain.

**Comment by blaqone** [comment]
Im only saying there is a problem when there are two because it tries to grab the other data.

**Comment by jamorham** [comment]
@blaqone this is an edge case that I haven't put any provision in to protect against as we're not expecting users to be running G6 and G7 at the same time on the same phone. If you want to test this then I think you need to move one of the sensors to another phone to be able to test it properly. To work around this I would have to make the user interface more complex to ask the user to choose which device to operate companion mode on and I don't really want to do that as it is quite simple and easy to use at the moment.

**Reply by blaqone** [reply]
I do t want anything to be changed. I only realized it and mentioned it. That’s all.
